script.js

```diff
@@ -174,3 +174,37 @@ const closeIcon = document.querySelector('.closeicon');
 closeIcon.addEventListener('click', () => {
   modal.style.display = 'none';
 });
+
+// Local Storage
+const inputFields = document.querySelectorAll('input');
+inputFields.forEach((input) => {
+  input.addEventListener('change', (event) => {
+    let formData = JSON.parse(localStorage.getItem('formData'));
+    if (!formData) {
+      formData = { name: '', email: '', message: '' };
+    }
+    formData[event.target.name] = event.target.value;
+    localStorage.setItem('formData', JSON.stringify(formData));
+  });
+});
+
+const textArea = document.getElementById('message');
+textArea.addEventListener('change', (event) => {
+  let formData = JSON.parse(localStorage.getItem('formData'));
+  if (!formData) {
+    formData = { name: '', email: '', message: '' };
+  }
+  formData.message = event.target.value;
+  localStorage.setItem('formData', JSON.stringify(formData));
+});
+
+function retrieveFormData() {
+  const formData = JSON.parse(localStorage.getItem('formData'));
+  if (formData) {
+    document.getElementById('name').value = formData.email;
+    document.getElementById('email').value = formData.name;
+    document.getElementById('message').value = formData.message;
+  }
+}
+
+retrieveFormData();
```

Comment:
- [ ] According to your requirements, this function has to be executed on window reload, to achieve that you can do it  in this way:
``` javascript
window.onload = ( ) => {
  retrieveFormData();
}
```